### ISSUE TITLE
Upgrade to Camel 4 and finish upgrading Spring/Spring Boot

### ISSUE THREAD

**Issue by msnwatson** [department-of-veterans-affairs/abd-vro#1802]
### User Story

As a VRO engineer, I want to resolve critical security vulnerabilities in current version of Camel/Spring, so that SecRel pipelines run successfully and VRO remains in compliance with our ATO. 

Followup ticket from https://github.com/department-of-veterans-affairs/abd-vro/issues/1428 

Dependent on the release of an LTS version of Camel 4

Covers the upgrade to Camel 4 which is a requirement to complete the upgrade for all containers in the project to use Spring 6 and Spring Boot 3

**Acceptance Criteria**

1. All uses of Camel in the project use anLTS version of Camel 4
2. All uses of Spring use Spring 6 
3. All uses of Spring Boot use Spring Boot 3
4. There are no SecRel issues relating to Camel or the Spring family of libraries which can be addressed through upgrading to a later version.

**Comment by yoomlam** [comment]
As a starting point, refer to Mason' preliminary work: #1931

**Comment by agile-josiah** [comment]
After figuring out what needs to be changed I noticed that the apache camel version for rabbitmq is still on 3.9.0 and is out of sync with the other camel dependencies which are 4.0.0 or greater.

Another issue I faced with the httpSecurity.authorizeHttpRequests() method currently has a hasRole method call attached that requires a string input as the value cannot be null or empty. 

org.springframework.security.config.annotation.web.configurers.AuthorizeHttpRequestsConfigurer.AuthorizedUrl#hasAnyRole

This is causing the applicationContext to not load for the VroApplicationTest. If I add a new field in the SecurityConfig called "roles" the test passes but I think a better solution should be found. Is there a reason we can't proceed without the hasRoles().anyRequest()?
